tests/4.5/application_kernels/declare_target_subroutine.F90

```diff
@@ -0,0 +1,56 @@
+!===--- declare_target_subroutine.F90 --------------------------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses a declare target construct to within a subroutine, called
+! on the device. It takes in an array argument and checks that the array
+! is summed correctly. This test was provided by the LLNL FGPU repository,
+! https://github.com/LLNL/FGPU/, derived from the OpenMP 4.5 examples.
+! Thanks to David Richards and Aaron Black for providing this test.
+!
+!===------------------------------------------------------------------------===//
+
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM declare_target_subroutine
```

Comment:
Add CONTAINS to move SUBROUTINE inside the program scope.

Reply:
Tried this, no change to results or the error message.

Reply:
Need to ping the authors about this test to see if anything has changed

Reply:
Also check with the examples document. Consider sending a note to vendors.

Reply:
"test_declare" needs an explicit interface (or be in a use-assocated module or being a contained procedure) as it uses a deferred-shape array argument.

If you don't want to make it a contained procedure or put into a module, you could use something like:
```Fortran
INTERFACE
  SUBROUTINE test_declare(a, sum)
    !$omp declare target(test_declare)
    INTEGER,POINTER,INTENT(in) :: a(:)
    INTEGER,INTENT(inout):: sum
  END SUBROUTINE
END INTERFACE
```

Reply:
Thanks @tob2! This resolves it.